engine/src/engine.rs

```diff
@@ -921,19 +978,19 @@ pub fn submit<I: IO + Copy, E: Env, P: PromiseHandler>(
         // TODO: charge for storage
     };
 
-    // Give refund
+    // Getting an amount of real gas usage after the execution of the transaction.
     let gas_used = match &result {
         Ok(submit_result) => submit_result.gas_used,
         Err(engine_err) => engine_err.gas_used,
     };
-    refund_unused_gas(&mut io, &sender, gas_used, prepaid_amount, &relayer_address).map_err(
-        |e| EngineError {
+    engine
```

Comment:
This solution looks completely non-obvious and potentially dangerous to me. The fact is that `charge_gas` existed (executed) before the transaction was EVM completed, and the user's balance changed before the transaction was completed in the EVM. Those. In a fact, Transfer, and Balance **SHOULD** work differently than before. Because their balance amount will be different. Previously this would have been an EVM error, now it is a NEAR level error. Such a decision seems controversial and unreasonable.